src/analysis/retail/monk/mistweaver/normalizers/EventLinks/VivifyEventLinks.ts

```diff
@@ -1,19 +1,64 @@
 import SPELLS from 'common/SPELLS';
 import { EventLink } from 'parser/core/EventLinkNormalizer';
 import { EventType } from 'parser/core/Events';
-import { VIVIFY, CAST_BUFFER_MS } from './EventLinkConstants';
+import {
+  VIVIFY,
+  CAST_BUFFER_MS,
+  ZEN_PULSE_VIVIFY,
+  VIVACIOUS_VIVIFICATION,
+  ZEN_PULSE_CONSUME,
+} from './EventLinkConstants';
+import { TALENTS_MONK } from 'common/TALENTS';
 
 export const VIVIFY_EVENT_LINKS: EventLink[] = [
   {
     linkRelation: VIVIFY,
     linkingEventId: [SPELLS.VIVIFY.id],
-    linkingEventType: [EventType.Cast, EventType.BeginChannel],
-    referencedEventId: [SPELLS.INVIGORATING_MISTS_HEAL.id, SPELLS.VIVIFY.id],
+    linkingEventType: [EventType.Heal],
+    referencedEventId: [SPELLS.INVIGORATING_MISTS_HEAL.id],
     referencedEventType: [EventType.Heal],
-    backwardBufferMs: CAST_BUFFER_MS,
+    backwardBufferMs: 50,
+    forwardBufferMs: 50,
```

Comment:
```suggestion
    backwardBufferMs: CAST_BUFFER_MS,
    forwardBufferMs: CAST_BUFFER_MS,
```
for consistency, also 50 might be a bit low cause of server lag :3

Reply:
50 was the result of testing and to fix the bug. I can put it into a constant but for now if we want but might have to tinker with this one a bit. 100 was too large and it was grabbing two different casts worth-- been happening ever since they changed VV to not be consumed if you gained the buff midcast